metaflow/system/metaflow_system_logger.py

```diff
@@ -0,0 +1,92 @@
+import sys
+from typing import Dict, Any, Optional, Union
+
+
+class SystemLogger(object):
+    def __init__(self):
+        self._logger = None
+        self._flow_name = None
+
+    def __del__(self):
+        if self._flow_name == "not_a_real_flow":
+            self.logger.terminate()
+
+    def init_environment_outside_flow(
```

Comment:
can you add comments here for future readers as to why/how the logger needs to be/is constructed in this manner?

Reply:
Will address